products/views.py

```diff
@@ -2,16 +2,106 @@
 Contains Views for products app
 """
 # pylint: disable=no-self-use, no-member
+from django.shortcuts import get_object_or_404, render
 from rest_framework import status, viewsets
-from rest_framework.authentication import TokenAuthentication
+from rest_framework.authentication import SessionAuthentication, TokenAuthentication
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
+from rest_framework.views import APIView
 
+from users.contants import CONTENT_MANAGER, ROLES, SYSTEM_ADMIN
+
+from .constants import ADD_PRODUCT, HOMEPAGE, PRODUCT_PAGE
 from .models import Product
 from .permissions import IsContentManager
 from .serializers import ProductSerializer
 
 
+class ProductHomePageView(APIView):
+    """
+    Renders the homepage of products
+    """
+
+    authentication_classes = [SessionAuthentication]
+
+    def get(self, request):
+        """
+        Renders the homepage template
+        Args:
+            request(HttpRequest):
+        Returns:
+            (render): Value containing template data to display
+        """
+        context = {
+            "products": Product.objects.all(),
+            "content_managers": [ROLES[CONTENT_MANAGER], ROLES[SYSTEM_ADMIN]],
```

Comment:
Instead of passing the roles list, we should just pass `can_manage_content` flag calculated within the backend.

Reply:
Done